core/src/ldml/ldml_transforms.cpp

```diff
@@ -831,6 +844,99 @@ transforms::load(
   return transforms;
 }
 
+// string
+
+// TODO-LDML: copypasta -> refactor
+std::u32string &normalize_nfd(std::u32string &str, UErrorCode &status) {
+  const icu::Normalizer2 *nfd = icu::Normalizer2::getNFDInstance(status);
```

Comment:
Could this function just `const std::u16string rstr = km::kbp::kmx::u32string_to_u16string(str);` 
then call the u16 version of normalize_nfd with rstr 
then have lines 864 to get the result back into the u32 str. ?

Reply:
yes, that's a good point. I was hoping it could do more of an in-place change, but  not for now.

Reply:
PTAL